### PR TITLE
Introduce GitHub Actions PR Check workflow, fixes JBIDE-27706

### PR DESCRIPTION
**Pull request by odockal**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by odockal** [comment]
Successful actions run on my fork: https://github.com/odockal/jbosstools-quarkus/pull/1/checks?check_run_id=2236610843

**Comment by odockal** [comment]
@sbouchet Ping, review please.

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        restore-keys: |
+          jbosstools-${{ runner.os }}-maven-
+
+    # Build and compile using Maven
+    - name: Build/Compile and run unit tests
+      uses: GabrielBB/xvfb-action@v1
+      with:
+        run: >
+          mvn clean verify -U -fae --settings build-ci/maven-settings.xml -DskipITests=true
+          -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true --quiet
```
Comment (sbouchet) [inline .github/workflows/build_test.yml:53]: please add --no-transfer-progress

**Review by sbouchet — CHANGES_REQUESTED** [review]

**Review by sbouchet — APPROVED** [review]
LGTM

**Pull request merged**